l3kernel/l3keys.dtx

```diff
@@ -679,6 +679,31 @@
 % \cs{l_keys_choice_tl} and \cs{l_keys_choice_int} in exactly
 % the same way as described for \texttt{.choices:nn}.
 %
+% \subsection{Key validity}
+%
+% Some keys will be used as settings which have a strictly limited scope
+% of validity. Some will be only available once, others will only be valid
+% until typesetting begins. To allow formats to support this in a structured
+% way, \pkg{l3keys} allows this information to be specified using the
+% \texttt{.validity:n} property.
+%
+% \begin{function}[added = 2021-11-26]{.usage:n}
+%   \begin{syntax}
+%     \meta{key} .usage:n = \meta{scope}
+%   \end{syntax}
+%   Defines the \meta{key} to have usage within the \meta{scope}, which
+%   should be one of \texttt{general}, \texttt{preamble} or \texttt{load}.
+% \end{function}
+%
+% \begin{variable}{\g_keys_usage_load_prop, \g_keys_usage_preamble_seq}
```

Comment:
Currently the documented names mismatch the implemented ones (starting in line 1653).

Reply:
Fixed. Opinion on whether I should go with two `prop`, one `clist` and one `prop` or the `prop`/`seq` split as now?

Reply:
For programming a `seq` is better than a `clist` (less overhead in mapping items). And since the load time options are package/module dependent a `prop` seems like a good choice. So my opinion is pretty much stick with the current variables.

Reply:
that very much depends on the viewpoint I would say. For expl3 based packages I agree but for 2e type packages a seq or a prop is not that helpful and we intend to cater for both types of class/packages here. My inclination would be therefore to go with clists (but it is after midnight so who knows what I think tomorrow). Overhead in processing is not really that much of an argument given that we talk about a handful of packages per doc with a handful of keys in the their lists.

Reply:
if the `\g_keys_usage_preamble_seq` is in any way made available to 2e package (ie not under its expl3 name) then I think a clist is better for it. But perhaps there are no good use case why that should be offered.

Reply:
It's no longer a `clist`: a `prop` for each area makes most sense.